test/test_term.py

```diff
@@ -42,6 +43,27 @@ def derivative(self, t, left=True):
     assert vf_prod.shape == (3,)
     assert tree_allclose(vf_prod, term.prod(vf, dx))
 
+    # `# type: ignore` is used for contrapositive static type checking as per:
+    # https://github.com/microsoft/pyright/discussions/2411#discussioncomment-2028001
+    _term_static: diffrax.ControlTerm[Array] = term
+    _term_static_contra: diffrax.ControlTerm[diffrax.LevyVal] = term  # type: ignore
```

Comment:
I think the variables can be just a `_` as they're unused:
```python
_: diffrax.ControlTerm[Array] = term
_: diffrax.ControlTerm[diffrax.LevyVal] = term
```

Reply:
I've just noticed that if you do something like this pyright will raise a `reportAssignmentType` on line one, as it infers the type of `_` based on the hint on line two.
```python
_: diffrax.ControlTerm[Array] = term
reveal_type(_) # diffrax.ControlTerm[diffrax.LevyVal]
_: diffrax.ControlTerm[diffrax.LevyVal] = term

# Simple fix
_: diffrax.ControlTerm[Array] = term
__: diffrax.ControlTerm[diffrax.LevyVal] = term
```